sdks/go/pkg/beam/core/runtime/exec/coder.go

```diff
@@ -145,8 +145,15 @@ func MakeElementEncoder(c *coder.Coder) ElementEncoder {
 		}
 
 	case coder.Timer:
+		tc := coder.SkipW(c).Components[0]
+		if coder.IsKV(tc) {
+			return &timerEncoder{
+				elm: MakeElementEncoder(tc.Components[0]),
+				win: MakeWindowEncoder(c.Window),
+			}
+		}
 		return &timerEncoder{
-			elm: MakeElementEncoder(c.Components[0]),
```

Comment:
3. Yeah, not extracting them early leads to issues at this stage where we only need the key, but we require a KV. Could lead to subtle, weird errors I think.

Reply:
Removing the KV check here and always sending key coder from exec/translate